client/src/test/java/io/pravega/client/connection/impl/RawClientTest.java

```diff
@@ -151,7 +150,8 @@ public void testExceptionHandling() throws ConnectionFailedException {
         assertTrue(future.isCompletedExceptionally());
         assertFutureThrows("The future should be completed exceptionally", future,
                 t -> t instanceof ConnectionFailedException);
-
+        rawClient.close();
```

Comment:
Do we need to create a new `RawClient`?

Reply:
Yes. In real use the raw client can't be used once the exception is thrown. The mock was previously allowing that and this test was taking advantage of it. Now it mimics how it would actually get used.